sympy/core/add.py

```diff
@@ -610,10 +612,12 @@ def _eval_is_nonpositive(self):
         from sympy.core.exprtools import _monotonic_sign
         if not self.is_number:
             c, a = self.as_coeff_Add()
-            if not c.is_zero and a.is_nonpositive:
+            if not c.is_zero:
                 v = _monotonic_sign(a)
                 if v is not None:
                     s = v + c
+                    if a.is_nonnegative and v.is_integer == True:
```

Comment:
@asmeurer i will add comments and more complicated test cases for ```is_nonnegative``` and ```is_nonpositive```. should i do ?